File: backend/email_service.py
import os
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
from dotenv import load_dotenv, find_dotenv

load_dotenv(find_dotenv())

# SMTP Configuration
SMTP_SERVER = os.getenv("SMTP_SERVER")
SMTP_PORT = os.getenv("SMTP_PORT", "587")
SMTP_USERNAME = os.getenv("SMTP_USERNAME")
SMTP_PASSWORD = os.getenv("SMTP_PASSWORD")
SENDER_EMAIL = os.getenv("SENDER_EMAIL", SMTP_USERNAME)

def send_email(to_email: str, subject: str, body: str) -> dict:
    """
    Send an email using SMTP.
    If credentials are missing, simulates the process by printing to console.
    """
    if not all([SMTP_SERVER, SMTP_USERNAME, SMTP_PASSWORD]):
        print("\n" + "="*50)
        print("[(SIMULATED EMAIL)]")
        print(f"To: {to_email}")
        print(f"Subject: {subject}")
        print(f"Body: {body}")
        print("="*50 + "\n")
        return {"status": "simulated", "message": "Email printed to console"}

    try:
        msg = MIMEMultipart()
        msg['From'] = SENDER_EMAIL
        msg['To'] = to_email
        msg['Subject'] = subject
        msg.attach(MIMEText(body, 'plain'))

        server = smtplib.SMTP(SMTP_SERVER, int(SMTP_PORT))
        server.starttls()
        server.login(SMTP_USERNAME, SMTP_PASSWORD)
        server.send_message(msg)
        server.quit()
        
        print(f"[OK] Email sent to {to_email}")
        return {"status": "success", "message": "Email sent successfully"}
    except Exception as e:
        print(f"[ERROR] Email Error: {str(e)}")
        return {"status": "error", "message": str(e)}

def send_otp_email(to_email: str, otp: str, name: str = "User") -> dict:
    """Convenience function to send OTP email"""
    subject = "Your Verification Code - AI PG Management"
    body = f"""
Hello {name},

Your verification code is: {otp}

This code will expire in 10 minutes. If you did not request this, please ignore this email.

Best regards,
AI PG Management Team
"""
    return send_email(to_email, subject, body)

File: auth-backend/utils/emailService.js
const nodemailer = require('nodemailer');
require('dotenv').config();

const transporter = nodemailer.createTransport({
  host: process.env.SMTP_HOST || process.env.SMTP_SERVER,
  port: process.env.SMTP_PORT || 587,
  secure: process.env.SMTP_SECURE === 'true', // true for 465, false for other ports
  auth: {
    user: process.env.SMTP_USER || process.env.SMTP_USERNAME,
    pass: process.env.SMTP_PASS || process.env.SMTP_PASSWORD,
  },
});

/**
 * Send OTP email to user
 * @param {string} email 
 * @param {string} otp 
 */
const sendOTPEmail = async (email, otp) => {
  const sender = process.env.SMTP_USER || process.env.SMTP_USERNAME || process.env.SENDER_EMAIL;
  const mailOptions = {
    from: `"Auth Service" <${sender}>`,
    to: email,
    subject: 'Your OTP Code',
    text: `Your OTP is ${otp}. It expires in 5 minutes.`,
    html: `
      <div style="font-family: Arial, sans-serif; max-width: 600px; margin: auto; padding: 20px; border: 1px solid #eee; border-radius: 10px;">
        <h2 style="color: #4f46e5; text-align: center;">Your Verification Code</h2>
        <p style="font-size: 16px; color: #333;">Hello,</p>
        <p style="font-size: 16px; color: #333;">Your OTP for login is:</p>
        <div style="background-color: #f3f4f6; padding: 20px; border-radius: 8px; text-align: center; margin: 20px 0;">
          <span style="font-size: 32px; font-weight: bold; letter-spacing: 5px; color: #111827;">${otp}</span>
        </div>
        <p style="font-size: 14px; color: #6b7280; text-align: center;">This code will expire in <b>5 minutes</b>.</p>
        <p style="font-size: 12px; color: #9ca3af; text-align: center; margin-top: 30px;">If you did not request this code, please ignore this email.</p>
      </div>
    `,
  };

  try {
    await transporter.sendMail(mailOptions);
    console.log(`OTP email sent to ${email}`);
  } catch (error) {
    console.error('Error sending OTP email:', error);
    throw new Error('Failed to send email');
  }
};

module.exports = { sendOTPEmail };
